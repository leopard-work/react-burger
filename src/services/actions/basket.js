export const ADD_TO_BASKET = 'ADD_TO_BASKET';
export const REMOVE_FROM_BASKET = 'REMOVE_FROM_BASKET';
export const SORT_BASKET = 'SORT_BASKET';
export const BASKET_CLEAR = 'BASKET_CLEAR';