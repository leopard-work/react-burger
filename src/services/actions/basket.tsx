import { ItemProps } from "../../utils/types";

export const ADD_TO_BASKET: "ADD_TO_BASKET" = "ADD_TO_BASKET";
export const REMOVE_FROM_BASKET: "REMOVE_FROM_BASKET" = "REMOVE_FROM_BASKET";
export const SORT_BASKET: "SORT_BASKET" = "SORT_BASKET";
export const BASKET_CLEAR: "BASKET_CLEAR" = "BASKET_CLEAR";

interface ADD_TO_BASKET_ACTION {
  readonly type: typeof ADD_TO_BASKET;
  item: ItemProps;
}
interface REMOVE_FROM_BASKET_ACTION {
  readonly type: typeof REMOVE_FROM_BASKET;
  item: ItemProps & { uuid: string };
}
interface SORT_BASKET_ACTION {
  readonly type: typeof SORT_BASKET;
  index: number;
}
export interface BASKET_CLEAR_ACTION {
  readonly type: typeof BASKET_CLEAR;
}

export type BASKET_ACTIONS =
  | ADD_TO_BASKET_ACTION
  | REMOVE_FROM_BASKET_ACTION
  | SORT_BASKET_ACTION
  | BASKET_CLEAR_ACTION;
