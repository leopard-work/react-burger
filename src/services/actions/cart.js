import {createOrder, loadIngredients} from "../../utils/api";

export const GET_ITEMS_REQUEST = 'GET_ITEMS_REQUEST';
export const GET_ITEMS_SUCCESS = 'GET_ITEMS_SUCCESS';
export const GET_ITEMS_FAILED = 'GET_ITEMS_FAILED';
export const ADD_TO_BASKET = 'ADD_TO_BASKET';
export const REMOVE_FROM_BASKET = 'REMOVE_FROM_BASKET';
export const CHECKOUT_REQUEST = 'CHECKOUT_REQUEST';
export const CHECKOUT_SUCCESS = 'CHECKOUT_SUCCESS';
export const CHECKOUT_FAILED = 'CHECKOUT_FAILED';
export const CLEAR_ORDER = 'CLEAR_ORDER';
export const VIEW_ITEM = 'VIEW_ITEM';
export const CLOSE_VIEW_ITEM = 'CLOSE_VIEW_ITEM';
export const SORT_BASKET = 'SORT_BASKET';
export const CHANGE_ACTIVE_TAB = 'CHANGE_ACTIVE_TAB';

export function getItems() {
    return function(dispatch) {
        dispatch({type: GET_ITEMS_REQUEST});
        loadIngredients().then((data) => {
            dispatch({type: GET_ITEMS_SUCCESS, items: data});
        }).catch(() => {
            dispatch({type: GET_ITEMS_FAILED});
        })
    };
}

export function checkOutSend(body) {
    return function(dispatch) {
        dispatch({type: CHECKOUT_REQUEST});
        createOrder(body).then((data) => {
            dispatch({type: CHECKOUT_SUCCESS, order: data});
        }).catch(() => {
            dispatch({type: CHECKOUT_FAILED});
        })
    }
}

export function addToBasket(payload) {

}