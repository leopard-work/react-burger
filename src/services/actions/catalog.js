import { loadIngredients } from "../../utils/api";

export const GET_ITEMS_REQUEST = "GET_ITEMS_REQUEST";
export const GET_ITEMS_SUCCESS = "GET_ITEMS_SUCCESS";
export const GET_ITEMS_FAILED = "GET_ITEMS_FAILED";
export const CHANGE_ACTIVE_TAB = "CHANGE_ACTIVE_TAB";

export function getItems() {
  // Загрузка меню
  return function (dispatch) {
    dispatch({ type: GET_ITEMS_REQUEST });
    loadIngredients()
      .then((data) => {
        dispatch({ type: GET_ITEMS_SUCCESS, items: data });
      })
      .catch(() => {
        dispatch({ type: GET_ITEMS_FAILED });
      });
  };
}
