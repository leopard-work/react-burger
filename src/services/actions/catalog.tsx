import { loadIngredients } from "../../utils/api";
import { Dispatch } from "redux";

export const GET_ITEMS_REQUEST: "GET_ITEMS_REQUEST" = "GET_ITEMS_REQUEST";
export const GET_ITEMS_SUCCESS: "GET_ITEMS_SUCCESS" = "GET_ITEMS_SUCCESS";
export const GET_ITEMS_FAILED: "GET_ITEMS_FAILED" = "GET_ITEMS_FAILED";
export const CHANGE_ACTIVE_TAB: "CHANGE_ACTIVE_TAB" = "CHANGE_ACTIVE_TAB";

interface GetItemsRequestAction {
  readonly type: typeof GET_ITEMS_REQUEST;
}
interface GetItemsFailedAction {
  readonly type: typeof GET_ITEMS_FAILED;
}
interface GetItemsSuccessAction {
  readonly type: typeof GET_ITEMS_SUCCESS;
}

type GetItemsActions =
  | GetItemsRequestAction
  | GetItemsFailedAction
  | GetItemsSuccessAction;

type GetItemsDispatch = Dispatch<GetItemsActions>;

export function getItems() {
  // Загрузка меню
  return function (dispatch: GetItemsDispatch) {
    dispatch({ type: GET_ITEMS_REQUEST });
    loadIngredients()
      .then((data) => {
        dispatch({ type: GET_ITEMS_SUCCESS, items: data });
      })
      .catch(() => {
        dispatch({ type: GET_ITEMS_FAILED });
      });
  };
}
