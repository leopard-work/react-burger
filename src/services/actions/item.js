export const VIEW_ITEM = 'VIEW_ITEM';
export const CLOSE_VIEW_ITEM = 'CLOSE_VIEW_ITEM';