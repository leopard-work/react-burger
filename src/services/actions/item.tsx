import { ItemProps } from "../../utils/types";

export const VIEW_ITEM: "VIEW_ITEM" = "VIEW_ITEM";
export const CLOSE_VIEW_ITEM: "CLOSE_VIEW_ITEM" = "CLOSE_VIEW_ITEM";

interface VIEW_ITEM_ACTION {
  readonly type: typeof VIEW_ITEM;
  readonly item: ItemProps;
}
interface CLOSE_VIEW_ITEM_ACTION {
  readonly type: typeof CLOSE_VIEW_ITEM;
}

export type VIEW_ITEM_ACTIONS = VIEW_ITEM_ACTION | CLOSE_VIEW_ITEM_ACTION;
