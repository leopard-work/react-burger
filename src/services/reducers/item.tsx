import { CLOSE_VIEW_ITEM, VIEW_ITEM, VIEW_ITEM_ACTIONS } from "../actions/item";
import { ItemProps } from "../../utils/types";

type itemState = {
  viewItemModalOpen: boolean;
  viewItemElement: ItemProps;
};

const initialState: itemState = {
  viewItemModalOpen: false,
  viewItemElement: {
    _id: "",
    name: "",
    type: "",
    proteins: 0,
    fat: 0,
    carbohydrates: 0,
    calories: 0,
    price: 0,
    image: "",
    image_mobile: "",
    image_large: "",
    __v: 0,
  },
};

export const itemReducer = (
  state = initialState,
  action: VIEW_ITEM_ACTIONS
) => {
  switch (action.type) {
    case VIEW_ITEM: {
      // Подробный просмотр товара
      return {
        ...state,
        viewItemElement: action.item,
        viewItemModalOpen: true,
      };
    }
    case CLOSE_VIEW_ITEM: {
      // Закрытие окна и удаление информации подробного просмотра товара
      return {
        ...state,
        viewItemElement: initialState.viewItemElement,
        viewItemModalOpen: false,
      };
    }
    default: {
      return state;
    }
  }
};
