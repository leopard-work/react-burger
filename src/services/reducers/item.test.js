import { CLOSE_VIEW_ITEM, VIEW_ITEM } from "../actions/item";
import { initialState, itemReducer } from "./item";
import renderer from "react-test-renderer";
import React from "react";

const testItem = {
  _id: "",
  name: "testItem",
  type: "",
  proteins: 0,
  fat: 0,
  carbohydrates: 0,
  calories: 0,
  price: 0,
  image: "",
  image_mobile: "",
  image_large: "",
  __v: 0,
  count: 0,
};

describe("Открытие окна ингредиента", () => {
  it("- исходное состояние", () => {
    expect(itemReducer(initialState, {})).toEqual(initialState);
  });
  it("- обработка VIEW_ITEM", () => {
    expect(
      itemReducer(undefined, {
        type: VIEW_ITEM,
        item: testItem,
      })
    ).toEqual({
      viewItemElement: testItem,
      viewItemModalOpen: true,
    });
    const tree = renderer
      .create(<>Получение товара для открытия в модальном окне</>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("- обработка CLOSE_VIEW_ITEM", () => {
    expect(
      itemReducer(initialState, {
        type: CLOSE_VIEW_ITEM,
      })
    ).toEqual({
      viewItemElement: initialState.viewItemElement,
      viewItemModalOpen: false,
    });
    const tree = renderer
      .create(<>Закрытие модального окна и удаление товара</>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
