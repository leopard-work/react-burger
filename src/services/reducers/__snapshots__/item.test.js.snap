// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Открытие окна ингредиента - обработка CLOSE_VIEW_ITEM 1`] = `"Закрытие модального окна и удаление товара"`;

exports[`Открытие окна ингредиента - обработка VIEW_ITEM 1`] = `"Получение товара для открытия в модальном окне"`;
